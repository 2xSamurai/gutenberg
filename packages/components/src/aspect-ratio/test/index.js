/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { AspectRatio } from '../index';

describe( 'props', () => {
	test( 'should render correctly', () => {
		const { container } = render(
			<AspectRatio>
				<img alt="Snow" />
			</AspectRatio>
		);
		expect( container.firstChild ).toMatchSnapshot();
	} );

	test( 'should add different ratio when provided', () => {
		const { container: withRatio } = render(
			<AspectRatio ratio={ 21 / 9 }>
				<img alt="Snow" />
			</AspectRatio>
		);
		const { container: defaultRatio } = render(
			<AspectRatio>
				<img alt="Snow" />
			</AspectRatio>
		);
		expect( withRatio.firstChild ).toMatchDiffSnapshot(
			defaultRatio.firstChild
		);
	} );

	test( 'should add different width when provided', () => {
		const { container: withWidth } = render(
			<AspectRatio ratio={ 21 / 9 } width={ '320px' }>
				<img alt="Snow" />
			</AspectRatio>
		);
		const { container: defaultWidth } = render(
			<AspectRatio ratio={ 21 / 9 }>
				<img alt="Snow" />
			</AspectRatio>
		);
		expect( withWidth.firstChild ).toMatchDiffSnapshot(
			defaultWidth.firstChild
		);
	} );
} );
