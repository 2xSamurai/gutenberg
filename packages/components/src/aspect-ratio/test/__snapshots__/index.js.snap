// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should add different ratio when provided 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -7,8 +7,8 @@
      alt="Snow"
      class="css-17i08y2-content"
    />
    <div
      aria-hidden="true"
-     class="css-q6czde-AspectRatioResizer-resizerClasses e8w5upx0"
+     class="css-nz8uzz-AspectRatioResizer-resizerClasses e8w5upx0"
    />
  </div>
`;

exports[`props should add different width when provided 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,7 +1,7 @@
  <div
-   class="components-aspect-ratio css-eji9eb-AspectRatioView-classes e8w5upx1"
+   class="components-aspect-ratio css-y2g68t-AspectRatioView-classes e8w5upx1"
    data-wp-c16t="true"
    data-wp-component="AspectRatio"
  >
    <img
      alt="Snow"
`;

exports[`props should render correctly 1`] = `
.emotion-0 {
  max-width: 100%;
  position: relative;
  width: 100%;
}

.emotion-2 {
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
}

.emotion-3 {
  height: 0;
  pointer-events: none;
  padding-bottom: 100%;
}

<div
  class="components-aspect-ratio emotion-0 emotion-1"
  data-wp-c16t="true"
  data-wp-component="AspectRatio"
>
  <img
    alt="Snow"
    class="emotion-2"
  />
  <div
    aria-hidden="true"
    class="emotion-3 emotion-4"
  />
</div>
`;
